reserved_keyword_checker/reserved_keyword_checker.py

```diff
@@ -0,0 +1,292 @@
+          application
+        - app_name: name of the Django app that contains the violation
+        - module_name: the file path of the Python module containing the violation
+        - model_name: name of the Django model that contains the violation
+        - field: the actual offending field name that triggered the violation
+        - source: is this module/model where the offending field was defined, or was it
+          inherited from another model.
+        """
+        model_name = self.model._meta.concrete_model.__name__
+        app_name = self.model._meta.app_label
+        if self.local_app:
+            app_source = "Local"
+        else:
+            app_source = "3rd party"
+        if not self.inherited:
+            source = "Defined here"
+        else:
+            source = "Inherited"
+        return "{},{},{},{},{},{},{}".format(
+            self.system, app_source, app_name, self.module_name, model_name, self.field, source
+        )
+
+    @property
+    def module_name(self):
+        """
+        The path to the module containing the reserved keyword violation
+        """
+        module_string = re.sub(r'\.', '/', self.model._meta.concrete_model.__module__)
+        return "{}.py".format(module_string)
+
+    @property
+    def inherited(self):
+        """
+        Return whether or not this violation is defined in a parent model of the model
+        in which it was found
+        """
+        return self.field not in [f.name for f in self.model._meta.local_fields]
+    
+    @property
+    def local_app(self):
+        """
+        Return whether or not this violation was found in a local source file or a third
+        party installed app
+        """
+        return os.path.isfile(self.module_name)
+
+
+class ConfigurationException(Exception):
+    pass
+
+
+class Config(object):
+
+
+    def __init__(self, reserved_keyword_config_file, override_file, report_path):
+        self.reserved_keyword_config = self.read_config_file(reserved_keyword_config_file)
+        if override_file:
+            self.overrides = self.read_config_file(override_file)
+        else:
+            self.overrides = {}
+        self.validate_override_config()
+        self.report_path = report_path
+        self.report_file = os.path.join(report_path, "reserved_keyword_report.csv")
+
+    @staticmethod
+    def read_config_file(config_file_path):
+        click.echo("Loading config file: {}".format(config_file_path))
+        try:
+            with io.open(config_file_path, 'r') as config_file:
+                config_dict = yaml.safe_load(config_file)
+        except:
+            raise ConfigurationException("Unable to load config file: {}".format(config_file_path))
+        if not config_dict:
+            raise ConfigurationException("Config file is empty: {}".format(config_file_path))
+        return config_dict
+
+    def validate_override_config(self):
+        field_path_regex = re.compile(r'([A-Z]\w*\.)(\w+)')
```

Comment:
Nit: can you use a non-regex validator?  For instance, the regex you wrote would match this invalid input, causing the subsequent logic to consider it valid: `Hello.world.foo`.

I'd prefer something like, e.g.
```py
try:
    app_name, model_name = 'Hello.world'.split('.')  # Raises ValueError if more than one period.
    if app_name and not app_name[0].isupper():
        raise ValueError('App name must be CamelCase.')
except ValueError:
    ...
```